src/components/TicTacToe.tsx

```diff
@@ -1,20 +1,26 @@
 import * as React from 'react';
 import Game from '../tic-tac-toe';
 import GameBoard from './GameBoard';
-
-type GameState = {
-  board?: Board;
-  boardSize?: number;
-  game?: Game;
-};
+import BoardSizeForm from './BoardSizeForm';
 
 class TicTacToe extends React.Component<object, GameState> {
   private game: Game = new Game();
 
   public state = {
     board: undefined,
     boardSize: undefined,
-    game: undefined,
+    error: false,
+  };
+
+  private validateBoardSizeInput = (size?: number): boolean => {
+    if (size === undefined) {
+      return true;
+    }
+    const isNotNull = size !== null;
+    const threeOrAbove = size >= 3;
+    const elevenOrBelow = size <= 10;
+    const isValidInput = threeOrAbove && elevenOrBelow && isNotNull;
```

Comment:
Two things to note here:

1) by naming/assigning these conditions, they'll always be evaluated (as opposed to short-circuiting)
2) There's likely some simplification that can be done here: for example, a number that is >=3 and <= 10 is by definition not null.

Also, if you're going to name these conditions I often find it useful to classify them by business rules; here I would use something like `const isValidInput = isPresent && isWithinRange;`